pkg/providers/provider.go

```diff
@@ -33,6 +33,8 @@ type Provider interface {
 	ValidateNewSpec(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error
 	GenerateMHC() ([]byte, error)
 	ChangeDiff(currentSpec, newSpec *cluster.Spec) *types.ComponentChangeDiff
+	ApplyCloudControllerTolerations(ctx context.Context, clusterSpec *cluster.Spec, cluster *types.Cluster) error
+	UpdateCloudControllerTolerations(ctx context.Context, oldClusterSpec v1alpha1.ClusterSpec, newClusterSpec v1alpha1.ClusterSpec, cluster *types.Cluster) error
```

Comment:
Could we just make this part of the `RunPostUpgrade` method?
And for `ApplyCloudControllerTolerations` you can just create a `RunPostCreate`. It seems more flexible

Reply:
Moving to RunPostUpgrade does not work. Moving the RunPostUpgrade function itself lso does not work. Was stuck at this step.

2021-11-10T07:17:23.467-0800	V3	Reconciling CRS managed resource	{"name": "vsphere-csi-controller-role"}
2021-11-10T07:17:23.467-0800	V4	Getting objects from CRS managed configmap	{"name": "vsphere-csi-controller-role"}